DoxyFiles/man/man3/Langmuir.3
.TH "Langmuir" 3 "Segunda, 3 de Outubro de 2022" "Version 1.0.0" "Isotherm++" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Langmuir \- Classe com a equacao da isoterma de \fBLangmuir\fP  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <Langmuir\&.h>\fP
.PP
Subclasse de \fBTwoParameters\fP e \fBIsothermTemplate< Langmuir >\fP\&.
.SS "Membros Públicos"

.in +1c
.ti -1c
.RI "\fBDefineIdentity\fP ('Langmuir', ID::Langmuir)"
.br
.RI "Definicao de \fBID\fP para esta classe para fins de identificacao de erros\&. "
.ti -1c
.RI "\fBLangmuir\fP ()=default"
.br
.RI "Construtora default\&. "
.ti -1c
.RI "\fBLangmuir\fP (const \fBLangmuir\fP &_orig)=default"
.br
.RI "Construtora de copia\&. "
.ti -1c
.RI "virtual \fB~Langmuir\fP ()=default"
.br
.RI "Destrutora\&. "
.ti -1c
.RI "\fBLangmuir\fP (const Real &_qmax, const Real &_k1)"
.br
.RI "Construtora com os parametros que definem a isoterma de \fBLangmuir\fP\&. "
.ti -1c
.RI "\fBLangmuir\fP & \fBoperator=\fP (const \fBLangmuir\fP &_orig)=default"
.br
.RI "Sobrecarga do operador =\&. "
.ti -1c
.RI "Real \fBQmax\fP () const"
.br
.RI "Funcao que informa o valor da capacidade maxima de adsorcao em [M /M ]\&. \&. "
.ti -1c
.RI "Real \fBK1\fP () const"
.br
.RI "Funcao que retorna o coeficiente associado a constante de equilibrio de \fBLangmuir\fP\&. "
.ti -1c
.RI "void \fBQmax\fP (const Real &_qmax)"
.br
.RI "Funcao para alterar a capacidade maxima de adsorcao em [M /M ]\&. \&. "
.ti -1c
.RI "void \fBK1\fP (const Real &_k1)"
.br
.RI "Funcao que altera o coeficiente associado a constante de equilibrio de \fBLangmuir\fP\&. "
.ti -1c
.RI "Real \fBQe\fP (const Real &_c, const Real &) const"
.br
.RI "Funcao que calcula a quantidade de sorcao no equilibrio\&. "
.ti -1c
.RI "Real \fBQe\fP (const Real &_c) const override"
.br
.ti -1c
.RI "virtual std::unique_ptr< \fBIsotherm\fP > \fBCloneImplementation\fP () const override"
.br
.in -1c
.SS "Outros membros herdados"
.SH "Descrição detalhada"
.PP 
Classe com a equacao da isoterma de \fBLangmuir\fP 

Classe com as equacoes da isoterma de \fBLangmuir\fP - \fBFreundlich\fP\&.
.PP
Isoterma com dois parametros, $ q_{max} $ e $ K_1 $, cujas formulas sao as seguintes: \\begin{align} [Q_e(C_e, T) = q_{max}\\frac{K_1C_e}{1+K_1C_e}] \\end{align} Um artigo de referencia pode ser encontrado \fCaqui\fP\&. 
.PP
\fBAutores\fP
.RS 4
Iasmim Barboza Storck 
.PP
Lara Botelho Brum 
.PP
Luan Rodrigues Soares de Souza 
.PP
Joao Flavio Vieira de Vasconcellos 
.RE
.PP
\fBVersão\fP
.RS 4
1\&.0 
.RE
.PP
\fBData\fP
.RS 4
2022 
.RE
.PP
\fBBug\fP
.RS 4
Nao ha bugs conhecidos\&.
.RE
.PP
.PP
\fBCopyright\fP
.RS 4
GNU Public License\&. 
.RE
.PP

.SH "Construtores e Destrutores"
.PP 
.SS "Langmuir::Langmuir ()\fC [default]\fP"

.PP
Construtora default\&. Uso: 
.PP
.nf
Langmuir  var;

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros para esta funcao 
.RE
.PP

.SS "Langmuir::Langmuir (const \fBLangmuir\fP & _orig)\fC [default]\fP"

.PP
Construtora de copia\&. Uso: 
.PP
.nf
Langmuir  var;
Langmuir  var1(var);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_orig\fP Variavel do tipo \fBLangmuir\fP original\&. 
.RE
.PP

.SS "Langmuir::Langmuir (const Real & _qmax, const Real & _k1)"

.PP
Construtora com os parametros que definem a isoterma de \fBLangmuir\fP\&. Uso: 
.PP
.nf
Real q1(1\&.0);
Real k1(1\&.0);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_qmax\fP Capacidade maxima de adsorcao em [M /M ]\&. 
.br
\fI_k1\fP Constante de equilibrio de \fBLangmuir\fP\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_qmax\fP <= 0\&. 
.br
\fI_k1\fP <= 0\&. 
.RE
.PP

.SH "Funções membros"
.PP 
.SS "virtual std::unique_ptr< \fBIsotherm\fP > Langmuir::CloneImplementation () const\fC [inline]\fP, \fC [override]\fP, \fC [virtual]\fP"

.PP
Implementa \fBIsotherm\fP\&.
.SS "Langmuir::DefineIdentity ('Langmuir', ID::Langmuir)"

.PP
Definicao de \fBID\fP para esta classe para fins de identificacao de erros\&. 
.SS "Real Langmuir::K1 () const\fC [inline]\fP"

.PP
Funcao que retorna o coeficiente associado a constante de equilibrio de \fBLangmuir\fP\&. Uso: 
.PP
.nf
Langmuir  var1(q1, k1);
Real k2 = var1\&.K1();

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor do coeficiente associado a constante de equilibrio de \fBLangmuir\fP\&. 
.RE
.PP

.SS "void Langmuir::K1 (const Real & _k1)\fC [inline]\fP"

.PP
Funcao que altera o coeficiente associado a constante de equilibrio de \fBLangmuir\fP\&. Uso: 
.PP
.nf
Langmuir  var1(q1, k1);
Real k2(2\&.0);
var1\&.K1(k2);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_k1\fP Novo valor do coeficiente associado a constante de equilibrio de \fBLangmuir\fP\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_k1\fP <= 0\&. 
.RE
.PP

.SS "\fBLangmuir\fP & Langmuir::operator= (const \fBLangmuir\fP & _orig)\fC [default]\fP"

.PP
Sobrecarga do operador =\&. Uso: 
.PP
.nf
Langmuir  var1(q1, k1);
Langmuir  var2 = var1;

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_orig\fP Variavel do tipo \fBLangmuir\fP original\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Copia de _orig\&. 
.RE
.PP

.SS "Real Langmuir::Qe (const Real & _c) const\fC [inline]\fP, \fC [override]\fP, \fC [virtual]\fP"

.PP
Reimplementa \fBIsotherm\fP\&.
.SS "Real Langmuir::Qe (const Real & _c, const Real &) const\fC [virtual]\fP"

.PP
Funcao que calcula a quantidade de sorcao no equilibrio\&. Uso: 
.PP
.nf
Langmuir  var1(q1, k1);
Real ce(1);
Real qe = var1\&.Qe(ce);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_c\fP Concentracao do soluto\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor da quantidade de sorcao no equilibrio\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_c\fP < 0\&. 
.RE
.PP

.PP
Implementa \fBIsotherm\fP\&.
.SS "Real Langmuir::Qmax () const\fC [inline]\fP"

.PP
Funcao que informa o valor da capacidade maxima de adsorcao em [M /M ]\&. \&. Uso: 
.PP
.nf
Langmuir  var1(q1, k1);
Real q1 = var1\&.Qmax();

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor da capacidade maxima de adsorcao em [M /M ]\&. 
.RE
.PP

.SS "void Langmuir::Qmax (const Real & _qmax)\fC [inline]\fP"

.PP
Funcao para alterar a capacidade maxima de adsorcao em [M /M ]\&. \&. Uso: 
.PP
.nf
Langmuir  var1(q1, k1);
Real q2(3\&.0);
var1\&.Qmax(q2);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_qmax\fP Novo valor da capacidade maxima de adsorcao em [M /M ]\&. \&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_qmax\fP <= 0\&. 
.RE
.PP


.SH "Autor"
.PP 
Gerado automaticamente por Doxygen para Isotherm++ a partir do código-fonte\&.


IsothermLib/src/Isotherm/TwoParameters/Langmuir.cpp
//==============================================================================
// Name        : Langmuir.cpp
// Author      : Luan Rodrigues Soares de Souza
//               Joao Flavio Vieira de Vasconcellos
//               Iasmim Barboza Storck
// Version     : 1.0
// Description : Classe com as equacoes da isoterma de Langmuir
//
// Copyright   : Copyright (C) <2022>  Joao Flavio Vasconcellos
//                                      (jflavio at iprj.uerj.br)
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//
//==============================================================================

//==============================================================================
// includes lib c++
//==============================================================================

#include <iostream>

//==============================================================================
// includes da lib IsothermLib
//==============================================================================

#include <Error/IsoException.h>
#include <Isotherm/TwoParameters/Langmuir.h>

IST_NAMESPACE_OPEN

//==============================================================================
// Variaveis estaticas
//==============================================================================

VecPairString       detailsLangmuir
    {
        PairString  ( "Qmax"
                    , "Capacidade maxima de adsorcao")
    ,   PairString  ( "K1"
                    , "Constante de equilibrio de Langmuir")
    };

template<>
VecPairString IsothermTemplate < Langmuir >::infoIsotherm = detailsLangmuir;

//==============================================================================
// Funcoes da classe
//==============================================================================


#undef  __FUNCT__
#define __FUNCT__ "Langmuir :: Langmuir (const Real&, const Real&)"
Langmuir :: Langmuir    (   const Real&     _qmax
                        ,   const Real&     _k1)
                        :   TwoParameters(_qmax, _k1)
{


    try {

            if (_qmax <= 0.0)  throw
                    IsoException    (   IST_LOC
                                    ,   className()
                                    ,   BadQmaxLEZero);

            if (_k1 <= 0.0)  throw
                    IsoException    (   IST_LOC
                                    ,   className()
                                    ,   BadK1LEZero);

    } catch (const IsoException& _isoExcept) {

        std::cout << _isoExcept << "\n";
        abort();

    }

    setup = true;

    }

//==============================================================================
// Construtores
//==============================================================================

#undef  __FUNCT__
#define __FUNCT__ "Real Langmuir ::  Qe (const Real&, const Real&) const "
Real Langmuir ::  Qe    (   const Real& _ce
                        ,   const Real&
                        ) const 
{

    try {

        if (!setup) throw
                IsoException    (   IST_LOC
                                ,   className()
                                ,   BadCoefficient);

        if (_ce <= 0.0)  throw
                IsoException    (   IST_LOC
                                ,   className()
                                , BadCeLTZero);

    } catch (const IsoException& _isoExcept) {

        std::cout << _isoExcept << "\n";
        abort();
    }

auto    ptrValue    = std::begin(coeffValue);
auto    auxi        = *(ptrValue + 1) * _ce;
auto    value       = *ptrValue * auxi / (1.0 + auxi);

    return  (value >= ZERO ? value : 0.0);

}

IST_NAMESPACE_CLOSE